docker/opbeans/java/Dockerfile

```diff
@@ -1,10 +1,24 @@
-FROM opbeans/opbeans-java:latest
+FROM maven:3.5.3-jdk-10
+# setting --build-arg JAVA_AGENT_BRANCH=<branch> causes the agent to be built from source
+# instead of using the agent which comes pre-built with opbeans-java:latest
+ARG JAVA_AGENT_REPO=elastic/apm-agent-java
+ARG JAVA_AGENT_BRANCH=
+
+RUN mkdir /agent \
+  # making sure there is at least one file to COPY (otherwise docker complains)
+  && touch /agent/ignore
+COPY build-agent.sh .
+# noop if JAVA_AGENT_BRANCH is not set
+RUN ./build-agent.sh "${JAVA_AGENT_REPO}" "${JAVA_AGENT_BRANCH}"
 
+FROM opbeans/opbeans-java:latest
+# replaces the /agent/elastc-apm-agent.jar if it has been built by build-agent.sh
+COPY --from=0 /agent/* /app
```

Comment:
This caused an error during build for me:

```
Step 8/11 : COPY --from=0 /agent/* /app
ERROR: Service 'opbeans-java' failed to build: When using COPY with more than one source file, the destination must be a directory and end with a /
```

Reply:
seems to depend on the docker version. it works on my machine. Suffixed `/app/` with a slash, that should work in both cases.